zcl-builtin/matter/data-model/chip/test-cluster.xml

```diff
@@ -83,6 +83,10 @@ limitations under the License.
     <item name="a" type="NestedStructList" array="true" optional="false"/>
   </struct>
 
+  <struct name="DeepNestedStructList">
```

Comment:
OK, but shouldn't we be testing that the boolean for "contains array" actually gets set for this struct?  Where are we testing that?

Reply:
The real question is: where are we testing `struct_contains_array` behavior

Wherever that test is, it should be expanded to also test that it applies transitively

This is the only reference I found to `struct_contains_array` - see https://github.com/project-chip/zap/pull/1501#issuecomment-2569920792

(these zap tests are incredibly hard to grok)

Reply:
`struct_contains_array` had no test coverage, that's why I couldn't find it.   Tests added now to both cover transitive as well as at root